frontend/src/components/ThemeSwitcher/ThemeSwitcher.js

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from 'react';
+import { BsSunFill } from 'react-icons/bs';
+import { FaMoon } from 'react-icons/fa';
+
+const ThemeSwitcher = () => {
+  const getInitialTheme = () => {
+    const savedTheme = localStorage.getItem('theme');
+    return savedTheme ? savedTheme : (window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light');
+  };
+
+  const [theme, setTheme] = useState(getInitialTheme);
```

Comment:
you called a function getInitialTheme as a variable not like a function this may cause error